server/bundles/io.cloudbeaver.service.ldap.auth/src/io/cloudbeaver/service/ldap/auth/ssl/LdapSslSocketFactory.java

```diff
@@ -0,0 +1,56 @@
+package io.cloudbeaver.service.ldap.auth.ssl;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.Socket;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLSocketFactory;
+
+/**
+ * This class implementation correspond JNDI api.
+ * Intention is creating isolated ssl socket factory for each user, not for general keystore
+ */
+public class LdapSslSocketFactory extends SSLSocketFactory {
+    private static final ThreadLocal<SSLSocketFactory> tlsFactory = new ThreadLocal<>();
```

Comment:
Why use thread local? What if LDAP-related processing will happen in another thread?